tests/baselines/reference/classStaticBlock22.errors.txt

```diff
@@ -0,0 +1,174 @@
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(16,10): error TS1109: Expression expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(19,5): error TS18037: Await expression cannot be used inside class static block.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(28,14): error TS1005: ':' expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(31,9): error TS18037: Await expression cannot be used inside class static block.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(31,14): error TS1109: Expression expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(37,10): error TS18037: Await expression cannot be used inside class static block.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(37,15): error TS1109: Expression expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(53,5): error TS18037: Await expression cannot be used inside class static block.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(53,10): error TS1109: Expression expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(54,15): error TS1003: Identifier expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(54,15): error TS18037: Await expression cannot be used inside class static block.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(54,20): error TS1109: Expression expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(87,17): error TS18037: Await expression cannot be used inside class static block.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(87,22): error TS1109: Expression expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(87,24): error TS1005: ';' expected.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(88,17): error TS18037: Await expression cannot be used inside class static block.
+tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(88,23): error TS1109: Expression expected.
+
+
+==== tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts (27 errors) ====
+    let await: "any";
+    class C {
+      static {
+        let await: any; // illegal, cannot declare a new binding for await
+        ~~~
+!!! error TS2304: Cannot find name 'let'.
+            ~~~~~
+!!! error TS1005: ';' expected.
+            ~~~~~
+!!! error TS18037: Await expression cannot be used inside class static block.
+                 ~
+!!! error TS1109: Expression expected.
+                   ~~~
+!!! error TS2693: 'any' only refers to a type, but is being used as a value here.
+      }
+      static {
+        let { await } = {} as any; // illegal, cannot declare a new binding for await

```

Comment:
Missing an error here. Is this what #44459 is supposed to fix?

Reply:
Yep.

Reply:
There don't seem to be any errors here after the merge. Is this something you're still working on?

Reply:
It's unexpected. I'll to try to find out soon.

Reply:
It's very strange. There's diagnostic messages at [playground](https://www.typescriptlang.org/play?alwaysStrict=false&ts=4.4.0-dev.20210622#code/IYZwngdgxgBAZgV2gFwJYHsL3emAKAShgG8AoGCmAGwFNkSZgB3YVegXxgF4TPRGIYUu1KlQkWIhQYsAI2AAnfETKUYUTCHrNW9HgEZho8dHhIoaTDHkAvZSXKUAbosYs23GIfZA). But nothing comes in test case.

Reply:
Oh I got it. It's caused by `await;`.

You can see this [playground](https://www.typescriptlang.org/play?#code/IYZwngdgxgBAZgV2gFwJYHsL3emAKAShgG8AoGCmKTEZGYAd2FToF4YBGU8yx55ANykAvqSA).

Seems some diagnostic message will disappear when there's a `await;`;

I don't know why but it seems could be track in another thread.

Reply:
Ah, this is because we handle `await;` as a parse error in `+Await`, but `let { await }` is handled in the binder: https://github.com/microsoft/TypeScript/blob/3c71d54782789b76c1514e9dc37b42ee6a3a29fe/src/compiler/binder.ts#L2185-L2193

I would split this into two tests, one that has parse errors, and one that has static errors produced by the binder. The easiest thing to do would be to just cut out the tests that should be failing but aren't into a new test file.

Reply:
Done.